src/pypgstac/src/pypgstac/pypgstac.py

```diff
@@ -119,6 +120,175 @@ def loadextensions(self) -> None:
                 except Exception:
                     pass
 
+    def load_queryables(
+        self,
+        file: str,
+        collection_ids: Optional[list[str]] = None,
+        delete_missing: Optional[bool] = False,
+    ) -> None:
+        """Load queryables from a JSON file.
+
+        Args:
+            file: Path to the JSON file containing queryables definition
+            collection_ids: Comma-separated list of collection IDs to apply the
+                            queryables to
+            delete_missing: If True, delete properties not present in the file.
+                            If collection_ids is specified, only delete properties
+                            for those collections.
+        """
+
+        # Read the queryables JSON file
+        queryables_data = None
+        for item in read_json(file):
+            queryables_data = item
+            break  # We only need the first item
+
+        if not queryables_data:
+            raise ValueError(f"No valid JSON data found in {file}")
+
+        # Extract properties from the queryables definition
+        properties = queryables_data.get("properties", {})
+        if not properties:
+            raise ValueError("No properties found in queryables definition")
+
+        conn = self._db.connect()
+        with conn.cursor() as cur:
+            with conn.transaction():
+                # Insert each property as a queryable
+                for name, definition in properties.items():
+                    # Skip core fields that are already indexed
+                    if name in (
+                        "id",
+                        "geometry",
+                        "datetime",
+                        "end_datetime",
+                        "collection",
+                    ):
+                        continue
+
+                    # Determine property wrapper based on type
+                    property_wrapper = "to_text"  # default
+                    if definition.get("type") == "number":
+                        property_wrapper = "to_float"
+                    elif definition.get("type") == "integer":
+                        property_wrapper = "to_int"
+                    elif definition.get("format") == "date-time":
+                        property_wrapper = "to_tstz"
+                    elif definition.get("type") == "array":
+                        property_wrapper = "to_text_array"
+
+                    # Determine index type (default to BTREE)
+                    property_index_type = "BTREE"
```

Comment:
I'm leery about always setting an index. Adding too many indexes can actually be far worse than not having an index in many situations. It will likely lead to worse performance to have an index on every possible field. It will definitely make any write operations much slower and will lead to memory contention in many instances on read operations. Perhaps if we add another option indexfields=[foo, bar] in order to define which fields to add indexes to???

Reply:
Fine. What should we put in `property_wrapper` and `property_index_type` for not having the index created?